lua/tests/indent/cpp_spec.lua

```diff
@@ -0,0 +1,62 @@
+local whole_file = require('nvim-treesitter.test_utils').indent_whole_file
+local new_line = require('nvim-treesitter.test_utils').indent_new_line
+local scan_dir = require('plenary.scandir').scan_dir
+local Path = require('plenary.path')
+
+local opts = {
+  tabstop = 4,
+  shiftwidth = 4,
+  softtabstop = 0,
+  expandtab = true,
+  filetype = 'cpp',
+}
+
+local get_name = function(file)
+  return Path:new(file):make_relative('lua/tests/indent')
+end
+
+describe('indent C++:', function()
+  describe('whole file:', function()
+    local files = scan_dir('lua/tests/indent/c');
+    vim.list_extend(files, scan_dir('lua/tests/indent/cpp'))
+
+    for _, file in ipairs(files) do
+      it(get_name(file), function()
+        whole_file(file, opts)
+      end)
+    end
+  end)
+
+  describe('new line:', function()
+    local run = function(file, spec, title)
+      title = title and title or tostring(spec.on_line)
+      it(string.format('%s[%s]', get_name(file), title), function()
+        new_line(file, spec, opts)
+      end)
+    end
```

Comment:
i think you could factor out this run function :)

Reply:
You mean to have it as top-level local function? Or to move it to `test_utils.lua`?

Reply:
test_utils it, it's common to every spec.lua file you wrote in some details which you could generate with a closure.